designer/server/src/main/scala/pl/touk/nussknacker/ui/process/processingtype/ScenarioParametersDeterminer.scala

```diff
@@ -15,24 +16,24 @@ object ScenarioParametersDeterminer {
       processingType: ProcessingType,
   ): ProcessingMode = {
     val componentsToProcessingMode = components.map { component =>
-      val allowedProcessingModes = component.implementation.allowedProcessingModes
+      val allowedProcessingModes = component.component.allowedProcessingModes
       component.id -> allowedProcessingModes
     }.toMap
 
     componentsToProcessingMode.values.toList match {
       // FIXME: some proper errors and tests for that
       case Nil => throw new IllegalStateException(s"Empty list of components for processing type: $processingType")
       case nonEmptyList =>
-        val intersection = nonEmptyList.foldLeft(ProcessingMode.all) {
-          case (acc, Some(componentProcessingModes)) => acc.intersect(componentProcessingModes)
-          case (acc, None)                           => acc
+        val intersection = nonEmptyList.foldLeft(ProcessingMode.values.toSet) { case (acc, allowedProcessingModes) =>
+          acc.intersect(allowedProcessingModes.toProcessingModes)
         }
         intersection.toList match {
           case oneMode :: Nil => oneMode
           // FIXME: some proper errors and tests for that
           case Nil =>
-            val componentsWithDefinedAllowedProcessingModes = componentsToProcessingMode.collect {
-              case (id, Some(modes)) => id -> modes
+            val componentsWithDefinedAllowedProcessingModes = componentsToProcessingMode.flatMap {
+              case (id, AllowedProcessingModes.SetOf(modes)) => id -> modes :: Nil
+              case (_, AllowedProcessingModes.All)           => Nil
```

Comment:
is this ok? we ignore the modes in the `All` case

Reply:
Yes, this is intentional. Such behaviour has been achieved implicitly by using `.collect`. Moreover, it looks like in this case we are only interested in components which are allowed in separated sets of processing modes. Components that are allowed in all processing modes don't collide with any set of allowed processing modes and taking them into consideration could introduce noise

Reply:
WDYT about leaving a comment? Because I'm not sure if it's clear why we skip modes in this case